lib/rails_response_dumper/runner.rb

```diff
@@ -112,6 +113,9 @@ def run_dumps
               }
             }
 
+            # These modules add nonstandard internal data, some of which also lacks default deterministic serialization.
+            dump[:request][:headers].delete_if { |key, _| key =~ /^(rack|action_dispatch|action_controller)\./ }
```

Comment:
Is there a more direct API for accessing only the HTTP headers and not this internal data?

Reply:
I was disappointed to find that there was no such API on these objects. But I just did more searching and found what they *do* give you, which can be used as a much better filter here, at least for the current use cases I'll push up that change in a moment.